src/universal/components/PhaseItemMasonry.js

```diff
@@ -146,6 +146,11 @@ class PhaseItemMasonry extends React.Component<Props> {
       const {x, y} = bestEl.getBoundingClientRect()
       setClosingTransform(atmosphere, itemId, {x, y})
     }
+    if (atmosphere.startDragQueue && atmosphere.startDragQueue.length) {
+      // reply the startDrag event that fired before we received the end drag event
+      const queuedStart = atmosphere.startDragQueue.shift()
+      queuedStart()
```

Comment:
this is just a robust version of handling case 1. i still prefer the locking until the ack comes back because without locking there are still other edge cases.